src/OpenTelemetry.Extensions.Hosting/OpenTelemetryServicesExtensions.cs

```diff
@@ -55,6 +56,24 @@ public static IServiceCollection AddOpenTelemetryTracing(this IServiceCollection
             return services.AddOpenTelemetryTracing(sp => builder.Build(sp));
         }
 
+        /// <summary>
+        /// Writes OpenTelemetry self diagnostics events to the logging system.
+        /// </summary>
+        /// <remarks>
+        /// This needs to be called before <see cref="AddOpenTelemetryTracing(IServiceCollection)"/> in order to capture startup errors.
+        /// </remarks>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        public static IServiceCollection AddOpenTelemetrySelfDiagnosticsLogging(this IServiceCollection services)
```

Comment:
@ejsmith Just curious why you didn't use the new DI extensions here. I'm guessing because that would work for TracerProviderBuilder but this is a kind of universal thing for tracing, metrics, & logging?

Reply:
@CodeBlanch that was the idea originally and I was waiting for it, but the problem is that if I use those then I don't capture otel startup issues in the logging because it's not setup until after startup happens.

Reply:
Ah order issue. I think in `TelemetryHostedService` the `TracerProvider` is created in `StartAsync` so if in `SelfDiagnosticsLoggingHostedService` you got everything up and running in the ctor instead of `StartAsync` it might work and capture everything. But that's more academic you don't have to change anything unless you want to.

Reply:
@CodeBlanch good idea. I changed it around and I like it better being an extension on the TraceProviderBuilder.